lib/dry/struct/class_interface.rb

```diff
@@ -25,22 +25,42 @@ def inherited(klass)
       # and modifies {.schema} accordingly.
       #
       # @param [Symbol] name name of the defined attribute
-      # @param [Dry::Types::Definition] type
+      # @param [Dry::Types::Definition, nil] type or superclass of nested type
       # @return [Dry::Struct]
+      # @yield
+      #   If a block is given, it will be evaluated in the context of
+      #   and new struct class, and set as a nested type for the given
+      #   name
       # @raise [RepeatedAttributeError] when trying to define attribute with the
       #   same name as previously defined one
       #
       # @example
       #   class Language < Dry::Struct
       #     attribute :name, Types::String
+      #     attribute :details, Dry::Struct do
```

Comment:
@AMHOL is `attribute :details, Dry::Struct do ... end` expected usage still? I thought we wouldn't need to pass a type like `Dry::Struct` when supplying nested attributes? It seems redundant.

Reply:
@timriley that's optional, but allows you to define the parent type of the nested attribute, useful for extending existing types or defining a base type with custom configuration.

Reply:
@AMHOL OK, this makes sense. That's quite handy! So I suppose if the user supplied a type here, none of the default auto-class-defining behaviour would kick in?

Reply:
It would still kick in as normal, the only thing that passing this argument changes is the superclass of the nested type that is defined, so for example, if I do the following:

```ruby
class Address < Dry::Struct
    attribute :city, Types::Coercible::String
    attribute :zip, Types::Coercible::String
end

class User < Dry::Struct
  attribute :name, Types::Coercible::String
  attribute :age, Types::Coercible::Int
  attribute :address, Address do
    attribute :line_1, Types::Coercible::String
    attribute :line_2, Types::Coercible::String
  end
end
```

It will still define `User::Address`, and `User::Address` will descend from `Address` and have the attributes `line_1`, `line_2`, `city`, `zip`, as if it were defined like:

```ruby
class User < Dry::Struct
  class Address < ::Address
    attribute :line_1, Types::Coercible::String
    attribute :line_2, Types::Coercible::String
  end
end
```

Reply:
Wonderful!